tools/firtool/firtool.cpp

```diff
@@ -360,6 +360,25 @@ static cl::opt<bool> stripDebugInfo(
     cl::desc("Disable source locator information in output Verilog"),
     cl::init(false), cl::cat(mainCategory));
 
+// Build mode options.
+enum BuildMode { BuildModeDebug, BuildModeRelease };
+static cl::opt<BuildMode> buildMode(
+    "O", cl::desc("Controls how much optimization should be performed."),
+    cl::values(clEnumValN(BuildModeDebug, "debug",
+                          "compile with only necessary optimizations"),
+               clEnumValN(BuildModeRelease, "release",
+                          "compile with optimizations")),
+    cl::init(BuildModeRelease), cl::callback([](const BuildMode &buildMode) {
```

Comment:
To expose this in `--help`, add it to the main category:
```suggestion
    cl::init(BuildModeRelease), cl::cat(mainCategory), cl::callback([](const BuildMode &buildMode) {
```

Reply:
If not specified, is the hook still invoked (using default value)?  Just wondering if there's a good way to ensure we don't end up with three "opt" levels: debug, release, and unspecified (if, as this grows, the defaults elsewhere get tweaked without updating this).

Reply:
Did you consider/what do you think about using `cl::Prefix` (for "-Odebug", while apparently still supporting "-O=debug" and "-O debug")? Asking because that's the common convention for "Optimization level" flags, although there it's often a single character value.  OTOH we don't use that style of cl::opt otherwise so 🤷 .  Just a thought :).

Reply:
> To expose this in --help, add it to the main category:
Added it to the main category.
> If not specified, is the hook still invoked (using default value)?
No :(.  This is going to be annoying to ensure correctness. I think in the future we might want to overhaul how we are processing command line options.
> Did you consider/what do you think about using `cl::Prefix`
Yes, but I arbitrarily decided not to because none of the rest of the command line options work that way

Reply:
Sounds good! :+1: .